great_expectations/experimental/datasources/spark_datasource.py

```diff
@@ -0,0 +1,229 @@
+from __future__ import annotations
+
+import copy
+import logging
+import pathlib
+import re
+from typing import TYPE_CHECKING, Dict, List, Optional, Pattern, Tuple, Type, Union
+
+import pydantic
+from typing_extensions import ClassVar, Literal
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.alias_types import PathStr
+from great_expectations.core.batch_spec import PathBatchSpec
+from great_expectations.experimental.datasources.interfaces import (
+    Batch,
+    BatchRequest,
+    BatchRequestOptions,
+    BatchSortersDefinition,
+    DataAsset,
+    Datasource,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.execution_engine import ExecutionEngine
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SparkDatasourceError(Exception):
+    pass
+
+
+# TODO: <Alex>ALEX_NOTE-1/27/2023: We did not add "_pandas" suffix (nor qualified class name) for CSVSparkAsset in PandasDatasource module (we might need to do that at a later date, or come up with a different, more general, solution altogether).</Alex>
```

Comment:
Same note as before - for any code comments, do they make more sense as Jira tickets or discussion points at arch review or in our Slack? We have many outdated comments in our codebase - I'm wary of adding more if they aren't necessary

Reply:
@cdkini Good idea -- I will create the JIRA ticket.  Thanks.

Reply:
@cdkini This has been done.  Thanks.

Reply:
@billdirks The JIRA ticket has been filed; the comment will be deleted (in next commit).  Thanks.